.github/actions/setup-env/action.yml

```diff
@@ -46,12 +46,12 @@ runs:
     # Node.js
     - name: Setup pnpm
       if: inputs.setup_nodejs == 'true'
-      uses: pnpm/action-setup@v2
+      uses: pnpm/action-setup@v3
```

Comment:
Not blocking for this PR, but do we really need a separate action for pnpm setup? Could we replace it with `corepack enable`, like it's done in, e.g., elk: https://github.com/elk-zone/elk/blob/main/.github/workflows/ci.yml

Reply:
Good point, corepack seems to work quite well for me locally so we can explore it in the action as well, but I would prefer to do that in a separate issue and PR.